pkg/component/nodemanagement/machinecontrollermanager/machine_controller_manager.go

```diff
@@ -437,9 +444,35 @@ func (m *machineControllerManager) Deploy(ctx context.Context) error {
 		return err
 	}
 
+	if m.values.AutonomousShoot && m.namespace != metav1.NamespaceSystem {
+		// This deploys the target cluster RBAC (e.g., for managing nodes and bootstrap tokens) in the bootstrap cluster
+		// because we configure machine-controller-manager with the in-cluster config as the target kubeconfig for now.
+		// TODO(timebertt): disable machine-controller-manager's target cluster interaction entirely
+		//  when https://github.com/gardener/machine-controller-manager/issues/994 has been implemented
+		return managedresources.CreateForSeed(ctx, m.client, m.namespace, managedResourceTargetName, false, data)
+	}
 	return managedresources.CreateForShoot(ctx, m.client, m.namespace, managedResourceTargetName, managedresources.LabelValueGardener, false, data)
 }
 
+// TargetKubeconfig returns the path to the target kubeconfig file depending on the shoot configuration.
+func TargetKubeconfig(autonomousShoot bool, controlPlaneNamespace string) string {
```

Comment:
Why is this function exported? It looks like it is only called within the package. If it should be used as a utility function it should be moved to a corresponding package.
WDYT?

Reply:
I kept the function exposed like the `ProviderSidecarContainer` function (which is a utility function, used by `controlplane` webhooks). If a given webhook implementation doesn't use `ProviderSidecarContainer`, it might still want to use the `TargetKubeconfig` function.

I don't have a strong opinion. That's just why I didn't put much thought into this :)

Reply:
I think all provider webhooks should use `ProviderSidecarContainer` to get the base definition of the container, at least. OTOH, I don't have a strong opinion on exporting this or not. I'm fine with both.

Reply:
Fair enough, unexported the function.